packages/runtime/src/templates/server.ts

```diff
@@ -36,11 +37,16 @@ const getNetlifyNextServer = (NextServer: NextServerType) => {
       return async (req, res, parsedUrl) => {
         // preserve the URL before Next.js mutates it for i18n
         const { url, headers } = req
-        // handle the original res.revalidate() request
-        await handler(req, res, parsedUrl)
-        // handle on-demand revalidation by purging the ODB cache
-        if (res.statusCode === 200 && headers['x-prerender-revalidate'] && this.netlifyConfig.revalidateToken) {
+        if (headers['x-prerender-revalidate'] && this.netlifyConfig.revalidateToken) {
+          // handle on-demand revalidation by purging the ODB cache
           await this.netlifyRevalidate(url)
+
+          res = res as unknown as BaseNextResponse
+          res.statusCode = 200
+          res.setHeader('x-nextjs-cache', 'REVALIDATED')
+          res.send()
+        } else {
+          await handler(req, res, parsedUrl)
```

Comment:
put it behind the flag in https://github.com/netlify/next-runtime/pull/2058/commits/30db86a2be7eea8673902708fb5f11d9f92bae39.

Reply:
When originally implementing this feature, i wasn't sure whether it was necessary to execute the original Next.js `handler` - it's not needed for the Netlify revalidate to work - but I left it in because it allows Next.js to validate the input and potentially throw an error, which will block the revalidate from happening. However, it was just a precaution (I can't think of a scenario where this would be necessary because our own revalidate API can also handle these errors) so i think it's fine to go with your rework. In fact, I don't think you need to put it behind the feature flag because the bulk of the change is already behind the `this.netlifyConfig.revalidateToken` condition, which itself is feature-flagged. You also don't need to `await` the original handler with that method, you can just return it (which is the how the default Next.js handler works).

Reply:
that's good context, thanks! If the on-demand revalidation stuff is still flagged, then we should definitely remove the second flag here to make things less complex.  I opened a follow-up: https://github.com/netlify/next-runtime/pull/2113. It's good to know we can